engine/src/main/java/org/terasology/math/SpiralIterable.java

```diff
@@ -0,0 +1,183 @@
+/*
+ * Copyright 2020 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.math;
+
+import com.google.common.base.Preconditions;
+import org.joml.Vector2i;
+import org.joml.Vector2ic;
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+
+/**
+ * An {@link Iterable} that iterates in
+ * a square-shapes spiral around the central point (inclusive).
+ * <br><br>
+ * The iteration starts in positive x direction.
+ * <br><br>
+ * The iterating vector is reused. <b>Do not attempt to store the instance</b> e.g. in a collection.
+ * @author Martin Steiger
+ */
+public class SpiralIterable implements Iterable<Vector2ic>{
```

Comment:
I think the documentation on this can be improved, but that's unrelated to this PR.